### PR TITLE
Merge dev branch into main

### PR DESCRIPTION
**Pull request by famfo**
Changes:
- Moved majority of printing to the screen out of `arch`

### CODE REVIEW

File: src/kernel/arch/x86_64/interrupts/interrupt_handler.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-use super::vga_buffer;
```
Comment (InfRandomness) [inline src/kernel/arch/x86_64/interrupts/interrupt_handler.rs:1]: Why ?

**Comment by InfRandomness** [comment]
I ain't sure about the secon conflict https://github.com/famfo/FrogOS/pull/6/conflicts.

**Comment by InfRandomness** [comment]
Do you approve the resolution to the conflicts?

File: src/kernel/arch/x86_64/interrupts/interrupt_handler.rs (hunk 1 above)
Comment (famfo) [reply src/kernel/arch/x86_64/interrupts/interrupt_handler.rs:1]: Its been moved to `kernel::vga_buffer`

**Comment by famfo** [comment]
Need to look at that at home, looks good for now.

**Pull request merged**